src/test/java/org/mockitousage/junitrule/StrictJUnitRuleTest.java

```diff
@@ -0,0 +1,182 @@
+package org.mockitousage.junitrule;
+
+import org.assertj.core.api.Assertions;
+import org.junit.After;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.Statement;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.exceptions.base.MockitoAssertionError;
+import org.mockito.internal.junit.JUnitRule;
+import org.mockito.internal.util.MockitoLogger;
+import org.mockitousage.IMethods;
+import org.mockitoutil.TestBase;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.*;
+
+public class StrictJUnitRuleTest extends TestBase {
+
+    private MockitoLogger logger = new MockitoLogger() {
+        public void log(Object what) {
+            throw new AssertionError("This implementation of rule should not warn about anything");
+        }
+    };
+
+    private JUnitRule jUnitRule = new JUnitRule(logger, JUnitRule.Strictness.STRICT_STUBS);
+
+    private FrameworkMethod dummy = mock(FrameworkMethod.class);
+
+    @After public void after() {
+        //so that the validate framework usage exceptions do not collide with the tests here
+        resetState();
+    }
+
+    @Test public void ok_when_no_stubbings() throws Throwable {
+        run(new MockitoStatement() {
+            public void evaluate(IMethods mock, IMethods mock2) throws Throwable {
+                mock.simpleMethod();
+                Mockito.verify(mock).simpleMethod();
+            }
+        });
+    }
+
+    @Test public void ok_when_all_stubbings_used() throws Throwable {
+        run(new MockitoStatement() {
+            public void evaluate(IMethods mock1, IMethods mock2) throws Throwable {
+                IMethods mock = mock(IMethods.class);
+                given(mock.simpleMethod(10)).willReturn("foo");
+                mock.simpleMethod(10);
+            }
+        });
+    }
+
+    @Test public void ok_when_used_and_mismatched_argument() throws Throwable {
+        run(new MockitoStatement() {
+            public void evaluate(IMethods mock, IMethods mock2) throws Throwable {
+                given(mock.simpleMethod(10)).willReturn("foo");
+                mock.simpleMethod(10);
+                mock.simpleMethod(15);
+            }
+        });
+    }
+
+    @Test public void fails_when_unused_stubbings() throws Throwable {
+        try {
```

Comment:
@szczepiq 
Would you mind using the ExpectedException rule, this would improve the readability and avoid try/fail/catch blocks which distracts the reader from the test case IMO.